File: Cpp/1441_Build_an_Array_With_Stack_Operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> result;
        int i = 1;
        for(auto& num : target){
            while(num != i){
                result.push_back("Push");
                result.push_back("Pop");
                ++i;
            }
            result.push_back("Push");
            ++i;
        }
        
        return result;
    }
};

File: Kotlin/1441_Build_an_Array_With_Stack_Operations.kt
class Solution {
    fun buildArray(target: IntArray, n: Int): List<String> {
        val result = ArrayList<String>()
        var i = 1
        
        target.forEach{ num ->
            while(num != i) {
                result.add("Push")
                result.add("Pop")
                ++i
            }
            result.add("Push")
            ++i
        }
        
        return result
    }
}